src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -390,11 +385,9 @@ void InferRequest::SetGraph(std::shared_ptr<Graph> graph) {
         IE_THROW(NetworkNotLoaded);
     }
 
-    if (!m_graph->GetNetwork()->is_dynamic()) {
-        allocate_inputs();
-        allocate_outputs();
-        variables_states_ = m_graph->AllocateVariablesMemories();
-    }
+    allocate_inputs();
+    allocate_outputs();
```

Comment:
BTW should we do this for dynamic shape? Because if the output blob is not exisitng, it will be allocated at wait() eventually.

Reply:
Yes, we need to crate at least empty blob, as users may take out blob before inference and save it somewhere, so after inference we need to update blob content only.

Reply:
Ah, in my branch I was doing it with calling SetBlob for empty size. But this looks better.